core/service/rollout_store.go

```diff
@@ -0,0 +1,131 @@
+	rolloutViewer, err := m.rolloutViewerDao.FindRolloutViewerByViewerIDAndRolloutID(ct, viewerID, m.rolloutID)
+	return &rolloutViewer.IsActivated, err
+}
+
+func (m *MaxViewersActivatorStore) SetIsActivated(ct context.Context, viewerID uint64, isActivated bool) *errs.Error {
+	rolloutViewer, err := m.rolloutViewerDao.FindRolloutViewerByViewerIDAndRolloutID(ct, viewerID, m.rolloutID)
+	if err != nil {
+		return err
+	}
+	rolloutViewer.IsActivated = isActivated
+	return m.rolloutViewerDao.UpdateRolloutViewer(ct, rolloutViewer)
+}
+
+func (m *MaxViewersActivatorStore) GetTotalViewers(ct context.Context, defaultViewers int) (int, *errs.Error) {
+	store, err := m.rolloutStoreDao.FindRolloutStoreByID(ct, m.rolloutID)
+	return store.TotalViewers, err
+}
+
+func (m *MaxViewersActivatorStore) SetTotalViewers(ct context.Context, totalViewers int) *errs.Error {
+	store, err := m.rolloutStoreDao.FindRolloutStoreByID(ct, m.rolloutID)
+	if err != nil {
+		return err
+	}
+	store.TotalViewers = totalViewers
+	return m.rolloutStoreDao.UpdateRolloutStore(ct, store)
+}
+
+func NewMaxViewersActivatorStore(rolloutViewerDao dao.RolloutViewer, rolloutStoreDao dao.RolloutStore, rolloutID uint64) *MaxViewersActivatorStore {
+	return &MaxViewersActivatorStore{rolloutID: rolloutID, rolloutViewerDao: rolloutViewerDao, rolloutStoreDao: rolloutStoreDao}
+}
+
+type ExperimentVersionSelectorStore struct {
+	rolloutViewerDao dao.RolloutViewer
+	rolloutID        uint64
+}
+
+var _ rollout.ExperimentVersionSelectorStore = (*ExperimentVersionSelectorStore)(nil)
+
+func (e *ExperimentVersionSelectorStore) GetViewerVersionNumber(ct context.Context, viewerID uint64) (*int, *errs.Error) {
+	viewer, err := e.rolloutViewerDao.FindRolloutViewerByViewerIDAndRolloutID(ct, viewerID, e.rolloutID)
+	return &viewer.VersionNumber, err
+}
+
+func (e *ExperimentVersionSelectorStore) SetViewerVersionNumber(ct context.Context, viewerID uint64, versionNumber int) *errs.Error {
+	viewer, err := e.rolloutViewerDao.FindRolloutViewerByViewerIDAndRolloutID(ct, viewerID, e.rolloutID)
+	if err != nil {
+		return err
+	}
+	viewer.VersionNumber = versionNumber
+	return e.rolloutViewerDao.UpdateRolloutViewer(ct, viewer)
+}
+
+func NewExperimentVersionSelectorStore(rolloutViewerDao dao.RolloutViewer, rolloutID uint64) *ExperimentVersionSelectorStore {
+	return &ExperimentVersionSelectorStore{rolloutViewerDao: rolloutViewerDao, rolloutID: rolloutID}
+}
+
+type PercentageActivatorStore struct {
+	rolloutViewerDao dao.RolloutViewer
+	rolloutID        uint64
+}
+
+var _ rollout.PercentageActivatorStore = (*PercentageActivatorStore)(nil)
+
+func (p *PercentageActivatorStore) GetIsActivated(ct context.Context, viewerID uint64) (*bool, *errs.Error) {
+	viewer, err := p.rolloutViewerDao.FindRolloutViewerByViewerIDAndRolloutID(ct, viewerID, p.rolloutID)
+	if err != nil {
+		return nil, err
+	}
+	return &viewer.IsActivated, nil
```

Comment:
```suggestion
	}
	
	return &viewer.IsActivated, nil
```
Return `nil, nil` if `err.Code` is `NotFound`.